lib/benchee/benchmark.ex

```diff
@@ -49,13 +56,32 @@ defmodule Benchee.Benchmark do
     %Suite{suite | run_times: run_times}
   end
 
-  @no_input :__no_input
-  @no_input_marker {@no_input, @no_input}
+  defp duplicated_job_name?(scenarios, job_name) do
+    Enum.any?(scenarios, fn(scenario) -> scenario.job_name == job_name end)
+  end
 
-  @doc """
-  Key in the result for when there were no inputs given.
-  """
-  def no_input, do: @no_input
+  defp add_scenario(suite = %Suite{jobs: jobs, scenarios: scenarios, configuration: config},
+                    job_name, function) do
+    new_scenarios = build_scenarios_for_job(job_name, function, config)
+    jobs = Map.put(jobs, job_name, function)
+    %Suite{suite | jobs: jobs, scenarios: List.flatten([scenarios | new_scenarios])}
+  end
+
+  defp build_scenarios_for_job(job_name, function, config)
+  defp build_scenarios_for_job(job_name, function, nil) do
+    [%Scenario{job_name: job_name, function: function, input: @no_input,
+               input_name: @no_input}]
+  end
```

Comment:
hm it's probably just some unit tests that are not used to not having the config in there, let's keep it like this for now I can later recheck how amny tests that'd be and how ok it'd be to fix them